In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '고속터미널역 8-1번, 8-2번 출구 사이']
df

자치구                     대여소명  대여소번호     년   월  시간     lambda  \
72000  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   0   8.500000   
72001  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   1   4.633333   
72002  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   2   2.266667   
72003  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   3   1.733333   
72004  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   4   1.233333   
72005  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   5   1.900000   
72006  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   6   3.566667   
72007  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   7   4.166667   
72008  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   8   4.866667   
72009  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6   9   5.566667   
72010  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  10   5.333333   
72011  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  11   6.433333   
72012  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  12   7.333333   
72013  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  13   8.666667   
72014  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  14  11.100000   
72015  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  15  13.166667   
72016  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  16  14.733333   
72017  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  17  24.933333   
72018  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  18  33.500000   
72019  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  19  34.600000   
72020  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  20  31.766667   
72021  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  21  32.233333   
72022  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  22  29.766667   
72023  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   6  23  16.400000   
72024  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   0   6.233333   
72025  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   1   2.966667   
72026  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   2   1.700000   
72027  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   3   0.866667   
72028  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   4   0.933333   
72029  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   5   0.966667   
72030  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   6   1.966667   
72031  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   7   3.566667   
72032  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   8   2.800000   
72033  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7   9   3.566667   
72034  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  10   3.366667   
72035  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  11   5.833333   
72036  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  12   5.533333   
72037  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  13   6.500000   
72038  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  14   9.500000   
72039  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  15  11.266667   
72040  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  16  13.333333   
72041  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  17  17.066667   
72042  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  18  24.400000   
72043  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  19  23.900000   
72044  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  20  24.566667   
72045  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  21  24.233333   
72046  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  22  20.866667   
72047  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   7  23  14.633333   
72048  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   0   2.500000   
72049  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   1   1.633333   
72050  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   2   1.033333   
72051  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   3   0.600000   
72052  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   4   0.733333   
72053  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   5   0.500000   
72054  서초구  고속터미널역 8-1번, 8-2번 출구 사이   2219  2020   8   6   0.933333   
72055  서초구  고속터미널역 8-1번, 8-2번 출

In [4]:
df = df.drop(['대여소번호', 'lambda', 'mu', 'c', 'rho', 'p_c', 'p_1+'], axis = 1)
df

자치구                     대여소명     년   월  시간           p_0           p_1  \
72000  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   0  8.977760e-02  8.324832e-02   
72001  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   1  1.374716e-01  1.194284e-01   
72002  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   2  1.543216e-01  1.311733e-01   
72003  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   3  3.679074e-02  3.751212e-02   
72004  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   4  2.336325e-03  2.881468e-03   
72005  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   5  3.151495e-09  7.810227e-09   
72006  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   6  5.630336e-04  7.530574e-04   
72007  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   7  3.162735e-02  3.267288e-02   
72008  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   8  9.258886e-03  1.047905e-02   
72009  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   9  1.509467e-03  1.909704e-03   
72010  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  10  2.753109e-03  3.362576e-03   
72011  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  11  6.152079e-02  5.966589e-02   
72012  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  12  3.915464e-03  4.681533e-03   
72013  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  13  2.919881e-02  3.036676e-02   
72014  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  14  2.385135e-02  2.521429e-02   
72015  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  15  7.727111e-02  7.301935e-02   
72016  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  16  2.105032e-01  1.664444e-01   
72017  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  17  4.739034e-02  4.720103e-02   
72018  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  18  4.265382e-02  4.291000e-02   
72019  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  19  3.245706e-02  3.345624e-02   
72020  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  20  5.008045e-02  4.961192e-02   
72021  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  21  9.856655e-03  1.110884e-02   
72022  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  22  2.174592e-03  2.693358e-03   
72023  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  23  2.626682e-02  2.755496e-02   
72024  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   0  1.821493e-01  1.493944e-01   
72025  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   1  1.416144e-01  1.223658e-01   
72026  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   2  1.691632e-02  1.835600e-02   
72027  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   3  2.919881e-02  3.036676e-02   
72028  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   4  1.104971e-05  1.819952e-05   
72029  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   5  1.689304e-06  3.061864e-06   
72030  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   6  8.053215e-04  1.055866e-03   
72031  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   7  4.293987e-03  5.105073e-03   
72032  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   8  2.907691e-03  3.539797e-03   
72033  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   9  8.766673e-04  1.143944e-03   
72034  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  10  1.357636e-01  1.182079e-01   
72035  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  11  1.552925e-04  2.227556e-04   
72036  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  12  3.005531e-02  3.118238e-02   
72037  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  13  3.843951e-02  3.904013e-02   
72038  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  14  2.665594e-02  2.792994e-02   
72039  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  15  2.376386e-01  1.813134e-01   
72040  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  16  5.552098e-02  5.443233e-02   
72041  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  17  5.630778e-02  5.512348e-02   
72042  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  18  5.723649e-02  5.593739e-02   
72043  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  19  3.968092e-02  4.018534e-02   
72044  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  20  2.378106e-02  2.514583e-02   
72045  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  21  1.762160e-02  1.906384e-02   
72046  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  22  1.232441e-02  1.367922e-02   
72047  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  23  1.774627e-02  1.918871e-02   
72048  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2

In [5]:
data = pd.read_csv('./고속터미널역 8-1번, 8-2번 출구 사이 균형, 불균형.csv', encoding = 'cp949')
data

자치구                     대여소명     년   월  시간        재고량      3사분위수  \
0    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   0  29.233333  20.316667   
1    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   1  28.600000  20.316667   
2    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   2  28.533333  20.316667   
3    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   3  28.966667  20.316667   
4    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   4  29.366667  20.316667   
5    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   5  30.533333  20.316667   
6    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   6  31.466667  20.316667   
7    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   7  31.600000  20.316667   
8    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   8  20.633333  20.316667   
9    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   9  21.800000  20.316667   
10   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  10  22.766667  20.316667   
11   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  11  22.566667  20.316667   
12   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  12  23.866667  20.316667   
13   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  13  24.200000  20.316667   
14   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  14  24.800000  20.316667   
15   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  15  24.033333  20.316667   
16   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  16  20.133333  20.316667   
17   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  17  20.033333  20.316667   
18   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  18  20.233333  20.316667   
19   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  19  21.266667  20.316667   
20   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  20  21.033333  20.316667   
21   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  21  24.700000  20.316667   
22   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  22  30.466667  20.316667   
23   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  23  31.266667  20.316667   
24   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   0  20.000000  20.316667   
25   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   1  19.700000  20.316667   
26   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   2  20.066667  20.316667   
27   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   3  20.300000  20.316667   
28   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   4  20.733333  20.316667   
29   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   5  21.333333  20.316667   
30   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   6  21.933333  20.316667   
31   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   7  22.666667  20.316667   
32   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   8  14.600000  20.316667   
33   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   9  15.433333  20.316667   
34   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  10  15.100000  20.316667   
35   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  11  16.900000  20.316667   
36   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  12  17.333333  20.316667   
37   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  13  17.433333  20.316667   
38   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  14  17.900000  20.316667   
39   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  15  14.466667  20.316667   
40   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  16  14.300000  20.316667   
41   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  17  13.966667  20.316667   
42   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  18  13.400000  20.316667   
43   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  19  13.733333  20.316667   
44   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  20  15.066667  20.316667   
45   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  21  16.900000  20.316667   
46   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  22  18.966667  20.316667   
47   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  23  20.066667  20.316667   
48   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   0  13.900000  20.316667   
49   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   1  13.433333  20.316667   
50   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   2  13.766667  20.316667   
51   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   3  14.200000  20.316667   
52   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   4  14.566667  20.316667   
53   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   5  14.900000  20.

In [6]:
data = data.drop(['3사분위수', '1사분위수'], axis = 1)
data

자치구                     대여소명     년   월  시간        재고량   균형
0    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   0  29.233333  불균형
1    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   1  28.600000  불균형
2    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   2  28.533333  불균형
3    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   3  28.966667  불균형
4    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   4  29.366667  불균형
5    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   5  30.533333  불균형
6    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   6  31.466667  불균형
7    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   7  31.600000  불균형
8    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   8  20.633333  불균형
9    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   9  21.800000  불균형
10   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  10  22.766667  불균형
11   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  11  22.566667  불균형
12   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  12  23.866667  불균형
13   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  13  24.200000  불균형
14   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  14  24.800000  불균형
15   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  15  24.033333  불균형
16   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  16  20.133333   균형
17   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  17  20.033333   균형
18   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  18  20.233333   균형
19   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  19  21.266667  불균형
20   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  20  21.033333  불균형
21   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  21  24.700000  불균형
22   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  22  30.466667  불균형
23   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  23  31.266667  불균형
24   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   0  20.000000   균형
25   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   1  19.700000   균형
26   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   2  20.066667   균형
27   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   3  20.300000   균형
28   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   4  20.733333  불균형
29   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   5  21.333333  불균형
30   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   6  21.933333  불균형
31   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   7  22.666667  불균형
32   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   8  14.600000   균형
33   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   9  15.433333   균형
34   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  10  15.100000   균형
35   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  11  16.900000   균형
36   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  12  17.333333   균형
37   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  13  17.433333   균형
38   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  14  17.900000   균형
39   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  15  14.466667   균형
40   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  16  14.300000   균형
41   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  17  13.966667   균형
42   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  18  13.400000  불균형
43   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  19  13.733333   균형
44   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  20  15.066667   균형
45   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  21  16.900000   균형
46   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  22  18.966667   균형
47   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  23  20.066667   균형
48   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   0  13.900000   균형
49   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   1  13.433333  불균형
50   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   2  13.766667   균형
51   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   3  14.200000   균형
52   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   4  14.566667   균형
53   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   5  14.900000   균형
54   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   6  14.966667   균형
55   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   7  15.633333   균형
56   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   8   9.500000  불균형
57   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   9  10.333333  불균형
58   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  10  10.966667  불균형
59   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  11  12.100000  불균형
60   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  12  12.566667  불균형
61   서초구  고속터미널역 8-1번, 8-2번 출구 사이  20

In [7]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구                     대여소명     년   월  시간           p_0           p_1  \
0    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   0  8.977760e-02  8.324832e-02   
1    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   1  1.374716e-01  1.194284e-01   
2    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   2  1.543216e-01  1.311733e-01   
3    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   3  3.679074e-02  3.751212e-02   
4    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   4  2.336325e-03  2.881468e-03   
5    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   5  3.151495e-09  7.810227e-09   
6    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   6  5.630336e-04  7.530574e-04   
7    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   7  3.162735e-02  3.267288e-02   
8    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   8  9.258886e-03  1.047905e-02   
9    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   9  1.509467e-03  1.909704e-03   
10   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  10  2.753109e-03  3.362576e-03   
11   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  11  6.152079e-02  5.966589e-02   
12   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  12  3.915464e-03  4.681533e-03   
13   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  13  2.919881e-02  3.036676e-02   
14   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  14  2.385135e-02  2.521429e-02   
15   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  15  7.727111e-02  7.301935e-02   
16   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  16  2.105032e-01  1.664444e-01   
17   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  17  4.739034e-02  4.720103e-02   
18   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  18  4.265382e-02  4.291000e-02   
19   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  19  3.245706e-02  3.345624e-02   
20   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  20  5.008045e-02  4.961192e-02   
21   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  21  9.856655e-03  1.110884e-02   
22   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  22  2.174592e-03  2.693358e-03   
23   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  23  2.626682e-02  2.755496e-02   
24   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   0  1.821493e-01  1.493944e-01   
25   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   1  1.416144e-01  1.223658e-01   
26   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   2  1.691632e-02  1.835600e-02   
27   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   3  2.919881e-02  3.036676e-02   
28   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   4  1.104971e-05  1.819952e-05   
29   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   5  1.689304e-06  3.061864e-06   
30   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   6  8.053215e-04  1.055866e-03   
31   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   7  4.293987e-03  5.105073e-03   
32   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   8  2.907691e-03  3.539797e-03   
33   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   9  8.766673e-04  1.143944e-03   
34   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  10  1.357636e-01  1.182079e-01   
35   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  11  1.552925e-04  2.227556e-04   
36   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  12  3.005531e-02  3.118238e-02   
37   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  13  3.843951e-02  3.904013e-02   
38   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  14  2.665594e-02  2.792994e-02   
39   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  15  2.376386e-01  1.813134e-01   
40   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  16  5.552098e-02  5.443233e-02   
41   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  17  5.630778e-02  5.512348e-02   
42   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  18  5.723649e-02  5.593739e-02   
43   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  19  3.968092e-02  4.018534e-02   
44   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  20  2.378106e-02  2.514583e-02   
45   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  21  1.762160e-02  1.906384e-02   
46   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  22  1.232441e-02  1.367922e-02   
47   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  23  1.774627e-02  1.918871e-02   
48   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   0  7.433168e-02  7.056805e-02   
49   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   1  2.4664

In [8]:
len(df_new)

288

In [9]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구                     대여소명     년   월  시간           p_0           p_1  \
0    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   0  8.977760e-02  8.324832e-02   
1    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   1  1.374716e-01  1.194284e-01   
2    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   2  1.543216e-01  1.311733e-01   
3    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   3  3.679074e-02  3.751212e-02   
4    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   4  2.336325e-03  2.881468e-03   
5    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   5  3.151495e-09  7.810227e-09   
6    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   6  5.630336e-04  7.530574e-04   
7    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   7  3.162735e-02  3.267288e-02   
8    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   8  9.258886e-03  1.047905e-02   
9    서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6   9  1.509467e-03  1.909704e-03   
10   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  10  2.753109e-03  3.362576e-03   
11   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  11  6.152079e-02  5.966589e-02   
12   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  12  3.915464e-03  4.681533e-03   
13   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  13  2.919881e-02  3.036676e-02   
14   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  14  2.385135e-02  2.521429e-02   
15   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  15  7.727111e-02  7.301935e-02   
16   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  16  2.105032e-01  1.664444e-01   
17   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  17  4.739034e-02  4.720103e-02   
18   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  18  4.265382e-02  4.291000e-02   
19   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  19  3.245706e-02  3.345624e-02   
20   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  20  5.008045e-02  4.961192e-02   
21   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  21  9.856655e-03  1.110884e-02   
22   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  22  2.174592e-03  2.693358e-03   
23   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   6  23  2.626682e-02  2.755496e-02   
24   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   0  1.821493e-01  1.493944e-01   
25   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   1  1.416144e-01  1.223658e-01   
26   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   2  1.691632e-02  1.835600e-02   
27   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   3  2.919881e-02  3.036676e-02   
28   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   4  1.104971e-05  1.819952e-05   
29   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   5  1.689304e-06  3.061864e-06   
30   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   6  8.053215e-04  1.055866e-03   
31   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   7  4.293987e-03  5.105073e-03   
34   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  10  1.357636e-01  1.182079e-01   
36   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  12  3.005531e-02  3.118238e-02   
37   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  13  3.843951e-02  3.904013e-02   
38   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  14  2.665594e-02  2.792994e-02   
39   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  15  2.376386e-01  1.813134e-01   
40   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  16  5.552098e-02  5.443233e-02   
41   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  17  5.630778e-02  5.512348e-02   
42   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  18  5.723649e-02  5.593739e-02   
43   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  19  3.968092e-02  4.018534e-02   
44   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  20  2.378106e-02  2.514583e-02   
45   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  21  1.762160e-02  1.906384e-02   
46   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  22  1.232441e-02  1.367922e-02   
47   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  23  1.774627e-02  1.918871e-02   
48   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   0  7.433168e-02  7.056805e-02   
49   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   1  2.466462e-01  1.859333e-01   
50   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   2  4.545455e-02  4.545455e-02   
54   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   6  2.014867e-01  1.611894e-01   
56   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   8  1.4280

In [10]:
len(z)

214

In [11]:
# TP

len(z[z['균형'] == '균형'])

109

In [12]:
# FN

len(z[z['균형'] == '불균형'])

105

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 105개

- 균형인데 균형이라고 판단한 수가 109개

In [13]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구                     대여소명     년   월  시간           p_0           p_1  \
32   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   8  2.907691e-03  3.539797e-03   
33   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7   9  8.766673e-04  1.143944e-03   
35   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   7  11  1.552925e-04  2.227556e-04   
51   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   3  2.993113e-04  4.144310e-04   
52   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   4  3.402246e-04  4.678089e-04   
53   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   5  6.683752e-05  1.002563e-04   
55   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   7  1.269582e-03  1.622243e-03   
57   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8   9  1.347435e-06  2.470298e-06   
58   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  10  3.278470e-04  4.517004e-04   
59   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  11  5.959715e-06  1.013151e-05   
60   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  12  2.590650e-03  3.175635e-03   
61   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  13  1.247316e-03  1.595404e-03   
62   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  14  1.154790e-02  1.287525e-02   
67   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  19  1.560535e-02  1.703329e-02   
71   서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020   8  23  6.090960e-03  7.084521e-03   
101  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10   5  4.867735e-14  2.068787e-13   
102  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10   6  2.682789e-04  3.736741e-04   
104  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10   8  2.235962e-02  2.375710e-02   
105  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10   9  4.232044e-04  5.750059e-04   
106  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10  10  7.188945e-03  8.272993e-03   
107  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10  11  3.419150e-02  3.508657e-02   
108  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10  12  5.993227e-03  6.978030e-03   
112  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10  16  2.511269e-02  2.643939e-02   
113  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  10  17  1.102789e-02  1.233443e-02   
122  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11   2  2.517210e-02  2.649695e-02   
125  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11   5  4.867735e-14  2.068787e-13   
126  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11   6  2.054436e-10  5.820903e-10   
127  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11   7  1.012463e-02  1.139020e-02   
128  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11   8  1.936850e-06  3.486330e-06   
129  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11   9  8.583623e-03  9.763871e-03   
130  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11  10  3.996316e-05  6.158257e-05   
131  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11  11  8.564019e-05  1.267920e-04   
132  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11  12  6.683752e-05  1.002563e-04   
143  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  11  23  1.599523e-02  1.742764e-02   
144  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   0  2.440665e-04  3.416931e-04   
145  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   1  1.610969e-05  2.602335e-05   
146  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   2  2.384186e-07  4.768373e-07   
149  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   5  6.373271e-11  1.911981e-10   
150  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   6  6.373271e-11  1.911981e-10   
152  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   8  1.557077e-01  1.321156e-01   
153  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12   9  3.745574e-06  6.520073e-06   
154  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  10  4.414335e-04  5.983876e-04   
155  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  11  1.491947e-04  2.144674e-04   
156  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  12  7.556744e-04  9.943084e-04   
157  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  13  6.641258e-04  8.801668e-04   
159  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  15  8.676317e-03  9.862288e-03   
160  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  16  2.954922e-03  3.593824e-03   
162  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  18  1.551318e-03  1.959560e-03   
166  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  22  5.298790e-02  5.219704e-02   
167  서초구  고속터미널역 8-1번, 8-2번 출구 사이  2020  12  23  1.0810

In [14]:
len(w)

74

In [15]:
# FP

len(w[w['균형'] == '균형'])

36

In [16]:
# TN

len(w[w['균형'] == '불균형'])

38

- 불균형인데 균형이라고 판단한 수는 36
- 불균형인데 불균형이라고 판단한 수는 38

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [17]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.5104166666666666

In [18]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.7517241379310344

In [19]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.5093457943925234